### PR TITLE
Cleanup code of Session and WebContents

### PR DESCRIPTION
**Pull request by zcbenz**
* Make BrowserWindow and WebContents share same `web-preferences`, no more duplicate code of dealing options of BrowserWindow and webview pages.
* Simplify the management of BrowserContext for different partitions.

Refs #959.

### CODE REVIEW

**Comment by deepak1556** [comment]
Awesome! :+1:

File: atom/browser/api/atom_api_web_contents.cc [hunk 1]
```diff
@@ -185,6 +196,11 @@ WebContents::WebContents(const mate::Dictionary& options) {
   AttachAsUserData(web_contents);
   InitWithWebContents(web_contents);
 
+  // Save the preferences.
+  base::DictionaryValue web_preferences;
+  mate::ConvertFromV8(isolate, options.GetHandle(), &web_preferences);
+  new WebContentsPreferences(web_contents, &web_preferences);
```
Comment (deepak1556) [inline atom/browser/api/atom_api_web_contents.cc]: why not a static method be used for this ?

Comment (zcbenz) [inline atom/browser/api/atom_api_web_contents.cc]: Either style (`new XXX` or `XXX:Create`) seems to be popular in Chromium, I just didn't bother to add a new static method.

**Comment by paulcbetts** [comment]
> Make BrowserWindow and WebContents share same web-preferences, no more duplicate code of dealing options of BrowserWindow and webview pages.

:clap:

File: atom/browser/api/atom_api_web_contents.cc (hunk 1 above)
Comment (deepak1556) [inline atom/browser/api/atom_api_web_contents.cc]: cool, thanks!

**Pull request merged**